apis/projectcontour/v1/httpproxy.go

```diff
@@ -101,17 +101,120 @@ type HeaderMatchCondition struct {
 	NotExact string `json:"notexact,omitempty"`
 }
 
+// ExtensionServiceReference names an ExtensionService resource.
+type ExtensionServiceReference struct {
+	// API version of the referent.
+	// If this field is not specified, the default "projectcontour.io/v1alpha1" will be used
```

Comment:
No, because we can't do defaulting in the CRD yet. This default will be enforced at the Contour level until we move to CRDv1. Once we've done that, this sort of behavior change is allowable, I think, under the GA guarantee, since we will be deprecating the v1alpha1 ExtensionService and replacing it with a v1 version.

Reply:
Also, this is now adding public APIs that folks will consume, so changes to this need to be maintained very carefully. This is why most of Contour is inside an `internal` package.

Reply:
Agreed, but I still think it's useful.

Reply:
In summary; I'm going to leave APIVersion here.

Reply:
Is Contour going to support all the versions that are configured? For example if `v1beta1` comes out, Contour will support both so users can switch? That only makes sense to me after the resource hits `v1`.